sdk/scheduler/src/main/java/com/mesosphere/sdk/storage/PersisterUtils.java

```diff
@@ -87,21 +93,24 @@ public static String getServiceNamespacedRootPath(String namespace, String pathN
     /**
      * Combines the provided path elements into a unified path, autocorrecting for any delimiters within the elements.
      *
-     * @param first The first element of the path
-     * @param second The second element of the path (within {@code first})
-     * @return {@code first/second}, with any duplicate slashes cleaned up
+     * @param paths Arbitrary number of paths to be combined
+     * @return All the paths combined, with any duplicate slashes cleaned up
      */
-    public static String join(final String first, final String second) {
-        if (first.endsWith(PATH_DELIM_STR) && second.startsWith(PATH_DELIM_STR)) {
-            // "hello/" + "/world" => "hello/world"
-            return new StringBuilder(first).deleteCharAt(first.length() - 1).append(second).toString();
-        } else if (first.endsWith(PATH_DELIM_STR) || second.startsWith(PATH_DELIM_STR)) {
-            // "hello/" + "world", OR "hello" + "/world" => "hello/world"
-            return new StringBuilder(first).append(second).toString();
-        } else {
-            // "hello" + "world" => "hello/world"
-            return new StringBuilder(first).append(PATH_DELIM).append(second).toString();
-        }
+    public static String join(final String... paths) {
+        return Arrays.stream(paths).reduce("", (first, second) -> {
```

Comment:
Question, what about using the Java `Paths.get()` method? Also, would it not also make sense to just strip the leading and trailing slashes and then join the strings.
* https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#strip-java.lang.String-java.lang.String-

Reply:
I had something like

Reply:
I am not sure what would break if I change. Just saw that we have a test case to check the same. I added a comment inside the renamed method adding an example of this.